### PR TITLE
Page refresh component

### PR DESCRIPTION
**Pull request by jlpadilla**
### Implementation
image

### Design
https://marvelapp.com/prototype/bd8dg5b/screen/75381887/handoff
image

### CODE REVIEW

File: src/AcmAutoRefreshSelect/AcmAutoRefreshSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        console.log('pollInterval', pollInterval)
+
+        const { refreshCookie, startPolling, stopPolling } = props
+        if (pollInterval === 0) {
+            stopPolling()
+        } else {
+            startPolling(pollInterval)
+        }
+        savePollInterval(refreshCookie, pollInterval)
+    }, [selected])
```
Comment (zlayne) [inline src/AcmAutoRefreshSelect/AcmAutoRefreshSelect.tsx:119]: might want to include a return here which is triggered on dismount?

File: src/AcmAutoRefreshSelect/AcmAutoRefreshSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    return (
+        <div className={classes.container}>
+            <div
+                className={classes.reloadButton}
+                tabIndex={0}
+                role={'button'}
+                onClick={() => {
+                    handleRefresh()
+                }}
+                onKeyPress={(e) => handleKeyPress(e)}
```
Comment (jlpadilla) [inline src/AcmAutoRefreshSelect/AcmAutoRefreshSelect.tsx:140]:
```suggestion
                onClick={handleRefresh}
                onKeyPress={handleKeyPress}
```

File: src/AcmAutoRefreshSelect/AcmAutoRefreshSelect.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+export const AutoRefreshSelect = () => (
+    <AcmPageCard>
+        <AcmAutoRefreshSelect
+            refetch={() => console.log('Refetch')}
+            startPolling={(pi) => console.log('startPolling', pi)}
+            stopPolling={() => console.log('stopPolling')}
+            pollInterval={pollInterval}
+            refreshValues={REFRESH_TIMES}
+            refreshCookie={OVERVIEW_REFRESH_INTERVAL_COOKIE}
```
Comment (jlpadilla) [inline src/AcmAutoRefreshSelect/AcmAutoRefreshSelect.stories.tsx:26]: I'd remove these 2 props (refreshValues, refreshCookie) to simplify the component.

If this gets reused by another component like App, I'd expect we want to keep the options and selected interval consistent across all ACM pages.

File: src/AcmAutoRefreshSelect/AcmAutoRefreshSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+const DEFAULT_REFRESH_TIME = 60
+
+export type AcmAutoRefreshSelectProps = {
+    refetch: () => void
+    refreshValues: number[]
+    refreshCookie: string
+    pollInterval: number
+    startPolling: (pollInterval: number) => void
+    stopPolling: () => void
```
Comment (jlpadilla) [inline src/AcmAutoRefreshSelect/AcmAutoRefreshSelect.tsx:14]: Do we really need `startPolling` and `stopPolling`?  It seems simpler to use only `refetch`.

Comment (RobDolares) [reply src/AcmAutoRefreshSelect/AcmAutoRefreshSelect.tsx:14]: I would think not - will try and work those out of component

File: src/AcmAutoRefreshSelect/AcmAutoRefreshSelect.tsx (hunk 1 above)
Comment (RobDolares) [reply src/AcmAutoRefreshSelect/AcmAutoRefreshSelect.tsx:119]: Opting to replace start/stop polling with refetch on interval.  But you are correct.

File: src/AcmAutoRefreshSelect/AcmAutoRefreshSelect.tsx [hunk 5]
```diff
@@ -0,0 +1,189 @@
+import React, { useEffect, useState } from 'react'
+import { SyncAltIcon } from '@patternfly/react-icons'
+import { Dropdown, DropdownItem, DropdownToggle } from '@patternfly/react-core'
+import { makeStyles } from '@material-ui/styles'
+
+const DEFAULT_REFRESH_TIME = 30
+const REFRESH_VALUES = [1, 30, 60, 5 * 60, 30 * 60, 0]
```
Comment (jlpadilla) [inline src/AcmAutoRefreshSelect/AcmAutoRefreshSelect.tsx:7]:
```suggestion
const DEFAULT_REFRESH_TIME = 60
const REFRESH_VALUES = [30, 60, 5 * 60, 30 * 60, 0]
```

File: src/AcmAutoRefreshSelect/AcmAutoRefreshSelect.tsx [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+    icon: {
+        maxWidth: '25px',
+        color: 'var(--pf-global--primary-color--100)',
+    },
+})
+
+export const getPollInterval = (OVERVIEW_REFRESH_INTERVAL_COOKIE: string) => {
+    let pollInterval = DEFAULT_REFRESH_TIME * 1000
+    const savedInterval = localStorage.getItem(OVERVIEW_REFRESH_INTERVAL_COOKIE)
+    console.log('savedInterval', savedInterval)
```
Comment (jlpadilla) [inline src/AcmAutoRefreshSelect/AcmAutoRefreshSelect.tsx:47]: Remove logging before merging.
```suggestion
```

File: src/AcmAutoRefreshSelect/AcmAutoRefreshSelect.tsx [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+            //
+        }
+    } else {
+        savePollInterval(OVERVIEW_REFRESH_INTERVAL_COOKIE, 30000)
+    }
+    return pollInterval
+}
+
+export const savePollInterval = (OVERVIEW_REFRESH_INTERVAL_COOKIE: string, pollInterval: number) => {
+    console.log('savepollinterval', pollInterval)
```
Comment (jlpadilla) [inline src/AcmAutoRefreshSelect/AcmAutoRefreshSelect.tsx:65]: Remove logging before merging.
```suggestion
```

**Comment by jlpadilla** [comment]
Approved. (I created the PR, so can't officially approve.)

**Review by RobDolares — APPROVED** [review]

**Pull request merged**